internal/temporal/worker_deployment.go

```diff
@@ -0,0 +1,208 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the MIT License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2024 Datadog, Inc.
+
+package temporal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/serviceerror"
+	temporalClient "go.temporal.io/sdk/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	temporaliov1alpha1 "github.com/DataDog/temporal-worker-controller/api/v1alpha1"
+)
+
+// VersionInfo contains information about a specific version
+type VersionInfo struct {
+	VersionID      string
+	Status         temporaliov1alpha1.VersionStatus
+	DrainedSince   *time.Time
+	RampPercentage float32
+	TaskQueues     []temporaliov1alpha1.TaskQueue
+	TestWorkflows  []temporaliov1alpha1.WorkflowExecution
+}
+
+// TemporalWorkerState represents the state of a worker deployment in Temporal
+type TemporalWorkerState struct {
+	DefaultVersionID     string
+	VersionConflictToken []byte
+	RampingVersionID     string
+	RampPercentage       float32
+	RampingSince         *metav1.Time
```

Comment:
Since we already have these fields, can we delete `VersionInfo.RampPercentage`?

Reply:
We could drop the field, but I think it's better modelling to keep it. A `RampPercentage` should only be set on the target version, and I think having `VersionInfo.RampPercentage` with no need to have access to the `TemporalWorkerState` is cleaner and more intuitive.

Reply:
My concern is about the possibility of failing to update the value correctly in both places, leading to undefined behavior if both values are later read by different parts of the program.

Maybe we could avoid the need for double bookkeeping while supporting a similar access pattern by adding a `GetRampPercentage() (float32, bool)` method to `TemporalWorkerState`?

Reply:
I'm not convinced this could race given how it's set (the per-version one is set based on the worker state one), but I've removed the per-version field anyway and just use the worker state field when required.